src/zhttpsocket.rs

```diff
@@ -292,11 +297,15 @@ impl SessionData {
         }
     }
 
-    fn add(&self, key: SessionKey, handle_index: usize) -> Result<Session, ()> {
+    fn add(&self, key: SessionKey, handle_index: usize) -> Result<Session, SessionAddError> {
         let inner = &mut *self.inner.lock().unwrap();
```

Comment:
mutex handling can be improved improved by handling the potential error that can occur when locking the mutex, instead of using unwrap()

Reply:
`lock()` fails if the mutex is poisoned, which means we have a logic error and so it's reasonable to abort the program on the spot